cpp/arcticdb/version/version_core.cpp

```diff
@@ -95,14 +96,17 @@ folly::Future<entity::AtomKey> async_write_dataframe_impl(
     InputTensorFrame&& frame,
     const WriteOptions& options,
     const std::shared_ptr<DeDupMap> &de_dup_map,
-    bool sparsify_floats
+    bool sparsify_floats,
+    bool ensure_sorted
     ) {
     ARCTICDB_SAMPLE(DoWrite, 0)
 
     // Slice the frame according to the write options
     frame.set_bucketize_dynamic(options.bucketize_dynamic);
     auto slicing_arg = get_slicing_policy(options, frame);
     auto partial_key = IndexPartialKey{frame.desc.id(), version_id};
+    sorting::check<ErrorCode::E_UNSORTED_DATA>(!ensure_sorted || frame.desc.get_sorted() == arcticc::pb2::descriptors_pb2::SortedValue::ASCENDING || !std::holds_alternative<stream::TimeseriesIndex>(frame.index),
```

Comment:
As commented above. We have to agree all together. We already discussed the names with Matt. Otherwise it is a back and forth in the PR reviews, which is not efficient.

Reply:
I'm not concerned about the naming honestly. I think this parameter is code for `sorted as far as ArcticDB is concerned`, and given this may change in the future (we should support descending data) I don't think we should be too fine-grained with the naming. If we name it `is_monotonic_ascending` we'd have to complicate matters once it supports descending data.

Reply:
But we cannot change what `ensure_sorted` means once this is merged as it's in the public Python API.

If we ever want to support querying descending data, we have to introduce yet another conflicting argument to the Python...